sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridConsumer.cs

```diff
@@ -70,16 +74,7 @@ private async Task<EventGridEvent[]> DeserializeEventGridEventsInternal(string r
             List<EventGridEvent> egEvents = new List<EventGridEvent>();
 
             // Deserialize raw JSON string into separate events, deserialize event envelope properties
-            MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes(requestContent));
-            JsonDocument requestDocument;
-            if (async)
-            {
-                requestDocument = await JsonDocument.ParseAsync(stream, default, cancellationToken).ConfigureAwait(false);
-            }
-            else
-            {
-                requestDocument = JsonDocument.Parse(stream, default);
-            }
+            JsonDocument requestDocument = await ParseJsonToDocument(requestContent, async, cancellationToken).ConfigureAwait(false);
```

Comment:
Should I also do an async vs sync check here? If so, I feel like that kind of makes the code messier than before I added a helper method.

Reply:
I don't think so - we still want to return a Task out of this method, and the caller will be responsible for calling EnsureCompleted if it is sync.